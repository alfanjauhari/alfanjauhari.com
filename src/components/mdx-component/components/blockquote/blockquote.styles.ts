import { styled } from '@/theme';

export const StyledBlockQuote = styled('blockquote', {
  fontStyle: 'italic',
  color: '$gray9',
  borderLeft: '2px solid $colors$gray4',
  my: '$5',
  pl: '$4',
});
