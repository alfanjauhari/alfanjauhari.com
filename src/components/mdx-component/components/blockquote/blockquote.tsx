import { BlockquoteHTMLAttributes } from 'react';
import { StyledBlockQuote } from './blockquote.styles';

export type BlockquoteProps = BlockquoteHTMLAttributes<HTMLElement>;

export function Blockquote(props: BlockquoteProps) {
  return <StyledBlockQuote {...props} />;
}
