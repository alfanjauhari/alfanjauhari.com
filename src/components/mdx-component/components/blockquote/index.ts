export * from './blockquote';
export * from './blockquote.styles';
