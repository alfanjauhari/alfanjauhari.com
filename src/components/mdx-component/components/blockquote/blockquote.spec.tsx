import { render, screen } from '@testing-library/react';
import { Blockquote } from './blockquote';

describe('MDXComponent/Blockquote', () => {
  it('should render successfuly', () => {
    render(
      <Blockquote data-testid="blockquote-component">
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Repudiandae
        vero error culpa, ut voluptate, asperiores doloremque nam autem neque
        rem voluptatibus vitae reiciendis doloribus odio libero quos esse
        provident numquam?
      </Blockquote>,
    );

    expect(screen.getByTestId('blockquote-component')).toBeVisible();
  });

  it('should match snapshot', () => {
    render(
      <Blockquote data-testid="blockquote-component">
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Repudiandae
        vero error culpa, ut voluptate, asperiores doloremque nam autem neque
        rem voluptatibus vitae reiciendis doloribus odio libero quos esse
        provident numquam?
      </Blockquote>,
    );

    expect(screen.getByTestId('blockquote-component')).toMatchSnapshot();
  });
});
