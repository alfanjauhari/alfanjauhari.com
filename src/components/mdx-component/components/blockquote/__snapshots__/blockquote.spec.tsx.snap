// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MDXComponent/Blockquote should match snapshot 1`] = `
<blockquote
  class="c-laoDFi"
  data-testid="blockquote-component"
>
  Lorem ipsum dolor sit amet consectetur adipisicing elit. Repudiandae vero error culpa, ut voluptate, asperiores doloremque nam autem neque rem voluptatibus vitae reiciendis doloribus odio libero quos esse provident numquam?
</blockquote>
`;
